include/admin_navigation.js

```diff
@@ -442,11 +533,12 @@ module.exports = function AdminNavigationModule(pb) {
             return navigation;
         }
 
-        AdminNavigation.additions = removeNode(id, AdminNavigation.additions);
+        AdminNavigation.additions[site] = removeNode(id, AdminNavigation.additions[site]);
 
-        util.forEach(AdminNavigation.childrenAdditions, function(node, parentId) {
-            AdminNavigation.childrenAdditions[parentId] = removeNode(id, node);
-        });
+        var childAdditionsMap = AdminNavigation.childrenAdditions[site];
+        for (var parentId in  childAdditionsMap) {
+            childAdditionsMap[parentId] = removeNode(id, childAdditionsMap[parentId]);
+        }
```

Comment:
This has been modified.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Saturday, July 11, 2015 at 7:12 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In include/admin_navigation.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34417141>:

>
> -        util.forEach(AdminNavigation.childrenAdditions, function(node, parentId) {
> -            AdminNavigation.childrenAdditions[parentId] = removeNode(id, node);
> -        });
> +        var childAdditionsMap = AdminNavigation.childrenAdditions[site];
> +        for (var parentId in  childAdditionsMap) {
> +            childAdditionsMap[parentId] = removeNode(id, childAdditionsMap[parentId]);
> +        }


Code climate likes to complain about this kind of "for" loop:

The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.


To solve this we created a utility function:

util.forEach(childAdditionsMap, function(value, key) {
  childAdditionsMap[key] = removeNode(id, value);
});


-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34417141>.